src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp

```diff
@@ -46,30 +46,90 @@
+        const auto src = image3D.pixels<Math::Vector<3, T>>()[slice];
+        for(std::size_t y = 0; y != src.size()[0]; ++y)
+            for(std::size_t x = 0; x != src.size()[1]; ++x)
+                dst[y][x] = Math::Vector3<T>{src[y][x]}; /* Alpha implicitly opaque */
+
+    } else if(channelCount == 2) {
+        const auto src = image3D.pixels<Math::Vector<2, T>>()[slice];
+        /* If the user didn't specify a custom swizzle, assume they want
+           the two channels compressed in separate slices, R in RGB and G
+           in Alpha. This significantly improves quality. */
+        if(!customSwizzle)
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::gather<'r', 'r', 'r', 'g'>(src[y][x]);
+        else
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::Vector3<T>::pad(src[y][x]); /* Alpha implicitly opaque */
+
+    } else if(channelCount == 1) {
+        const auto src = image3D.pixels<Math::Vector<1, T>>()[slice];
+        /* If the user didn't specify a custom swizzle, assume they want
+           a gray-scale image. Alpha is always implicitly opaque. */
+        if(!customSwizzle)
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::gather<'r', 'r', 'r'>(src[y][x]);
+        else
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::Vector3<T>::pad(src[y][x]); /* Alpha implicitly opaque */
+
+    } else CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+}
+
 template<UnsignedInt dimensions> Containers::Optional<Containers::Array<char>> convertLevelsToData(Containers::ArrayView<const BasicImageView<dimensions>> imageLevels, const Utility::ConfigurationGroup& configuration, ImageConverterFlags flags, BasisImageConverter::Format fileFormat) {
     /* Check input */
     const PixelFormat pixelFormat = imageLevels.front().format();
-    bool isSrgb;
+    bool isSrgb = false;
+    bool isHdr = false;
     switch(pixelFormat) {
         case PixelFormat::RGBA8Unorm:
         case PixelFormat::RGB8Unorm:
         case PixelFormat::RG8Unorm:
         case PixelFormat::R8Unorm:
-            isSrgb = false;
             break;
         case PixelFormat::RGBA8Srgb:
         case PixelFormat::RGB8Srgb:
         case PixelFormat::RG8Srgb:
         case PixelFormat::R8Srgb:
             isSrgb = true;
             break;
+        #if BASISU_LIB_VERSION >= 150
+        case PixelFormat::R32F:
+        case PixelFormat::RG32F:
+        case PixelFormat::RGB32F:
+        case PixelFormat::RGBA32F:
+            isHdr = true;
+            break;
+        #endif
```

Comment:
It doesn't support half floats as input? Sad :(

Reply:
Do we want to add that to the plugin as a convenience? We're already doing a bunch of handholding that the command line utility does (but not the API), like implicit swizzling, so I think it'd make sense.

Reply:
As we discussed elsewhere, I'd rather provide a dedicated utility for the most common swizzles and format conversions instead, usable with any plugins. Which will then be exposed in the `magnum-imageconverter` utility, or available to be called from C++ / Python directly.